### PR TITLE
Upgrade visualization

### PR DESCRIPTION
**Pull request by ale90**
Hi,

I upgraded the visualization to make it like Google original project. 

Bye
Ale90

### CODE REVIEW

**Comment by zylo117** [comment]
Hi, thanks for contributing. Is it possible to do this without importing extra packages, like using random color instead of webcolor?

**Comment by ale90** [comment]
Hi, 
my goal was not use random colors but  to get a view as close as possible to Google original project. For this reason I used the color list of the names used by Google and converted using webcolors.

File: efficientdet_test.py [hunk 1]
```diff
@@ -15,7 +16,7 @@
 from efficientdet.utils import BBoxTransform, ClipBoxes
 from utils.utils import preprocess, invert_affine, postprocess
 
-compound_coef = 0
+compound_coef = 7
```
Comment (zylo117) [inline efficientdet_test.py:19]: compound_coef = 0

File: efficientdet_test.py [hunk 2]
```diff
@@ -26,7 +27,7 @@
 threshold = 0.2
 iou_threshold = 0.2
 
-use_cuda = True
+use_cuda = False
```
Comment (zylo117) [inline efficientdet_test.py:30]: use_cuda = True

File: efficientdet_test.py [hunk 3]
```diff
@@ -41,7 +42,44 @@
...
+    result=(rgb_color.blue,rgb_color.green,rgb_color.red)
+    return result
+
+def standard_to_bgr(list_color_name):
+    standard= []
+    for i in range(len(list_color_name)-36): #-36 used to match the len(obj_list)
+        standard.append(from_colorname_to_bgr(list_color_name[i]))
+    return standard
+
+color_list = standard_to_bgr(STANDARD_COLORS)
```
Comment (zylo117) [inline efficientdet_test.py:82]: move STANDARD_COLORS & from_colorname_to_bgr & standard_to_bgr to utils/utils.py

File: efficientdet_test.py [hunk 4]
```diff
@@ -77,31 +115,45 @@
...
+        s_size = cv2.getTextSize(str('{:.0%}'.format(score)),0, fontScale=float(tl) / 3, thickness=tf)[0]
+        t_size = cv2.getTextSize(label, 0, fontScale=float(tl) / 3, thickness=tf)[0]
+        c2 = c1[0] + t_size[0]+s_size[0]+15, c1[1] - t_size[1] -3
+        cv2.rectangle(img, c1, c2 , color, -1)  # filled
+        cv2.putText(img, '{}: {:.0%}'.format(label, score), (c1[0],c1[1] - 2), 0, float(tl) / 3, [0, 0, 0], thickness=tf, lineType=cv2.FONT_HERSHEY_SIMPLEX)
+
+def get_index_label(label):
+    index = int(obj_list.index(label))
+    return index
+
```
Comment (zylo117) [inline efficientdet_test.py:134]: move this part to utils/utils.py

File: efficientdet_test.py [hunk 5]
```diff
@@ -77,31 +115,45 @@
...
 def display(preds, imgs, imshow=True, imwrite=False):
     for i in range(len(imgs)):
         if len(preds[i]['rois']) == 0:
             continue
 
         for j in range(len(preds[i]['rois'])):
-            (x1, y1, x2, y2) = preds[i]['rois'][j].astype(np.int)
-            cv2.rectangle(imgs[i], (x1, y1), (x2, y2), (255, 255, 0), 2)
+            x0, y0, x1, y1 = preds[i]['rois'][j].astype(np.int)
```
Comment (zylo117) [inline efficientdet_test.py:141]: I think it should stick to x1 y1 instead of x0 y1

File: efficientdet_test.py [hunk 6]
```diff
@@ -77,31 +115,45 @@
...
             score = float(preds[i]['scores'][j])
+            plot_one_box(imgs[i], [x0, y0, x1, y1], label=obj,score=score,color=color_list[get_index_label(obj)])
 
-            cv2.putText(imgs[i], '{}, {:.3f}'.format(obj, score),
-                        (x1, y1 + 10), cv2.FONT_HERSHEY_SIMPLEX, 0.5,
-                        (255, 255, 0), 1)
 
         if imshow:
-            cv2.imshow('img', imgs[i])
+            cv2.imshow('img',imgs[i])
```
Comment (zylo117) [inline efficientdet_test.py:148]: don't change this line

File: efficientdet_test.py [hunk 7]
```diff
@@ -77,31 +115,45 @@
...
             cv2.waitKey(0)
 
         if imwrite:
             cv2.imwrite(f'test/img_inferred_d{compound_coef}_this_repo_{i}.jpg', imgs[i])
 
 
 out = invert_affine(framed_metas, out)
-display(out, ori_imgs, imshow=False, imwrite=True)
+display(out, ori_imgs, imshow=True, imwrite=False)
```
Comment (zylo117) [inline efficientdet_test.py:156]: don't modify this line

**Comment by ale90** [comment]
Hi,

refactor done.

File: efficientdet_test.py (hunk 4 above)
Comment (zylo117) [reply efficientdet_test.py:94]: does plot_one_box have to be here instead of utils.py?

**Comment by zylo117** [comment]
does plot_one_box have to be here instead of utils.py?

**Comment by ale90** [comment]
Hi,

I moved plot_one_box to utils.py.

**Comment by zylo117** [comment]
ok, thank you

**Pull request merged**